src/web/mjs/connectors/Allanimesite2.mjs

```diff
@@ -0,0 +1,102 @@
+    }
+    get icon() {
+        return '/img/connectors/allanimesite';
+    }
+    canHandleURI(uri) {
+        return /(www\.)?allanime\.site\/anime/.test(uri);
+    }
+    async _getMangasFromPage(page) {
+        const uri = new URL(this.path + this.varQueryMangas.replace('%PAGE%', page), this.url);
+        const request = new Request(uri, this.requestOptions);
+        const data = await this.fetchJSON(request);
+        if (data.data == null) {
+            return {};
+        }
+        return data.data.shows.edges.map(element => {
+            return {
+                id: '/anime/'+element._id,
+                title: element.englishName ? element.englishName : element.name,
+            };
+        });
+    }
+    async _getChapters(manga) {
+        const request = new Request(new URL(manga.id, this.url), this.requestOptions);
+        let script = `
+        new Promise(resolve => {
+            let pages = __NUXT__;
+            resolve(pages);
+        });
+        `;
+        let data = await Engine.Request.fetchUI(request, script);
+        let chapterlist = [];
+        const mangaid = data.fetch[0].show._id;
+        let subchapters = data.fetch[0].show.availableEpisodesDetail.sub;
+        subchapters.forEach(chapter => {
+            chapterlist.push({
+                id : '/watch/'+mangaid+'/episode-'+chapter+'-sub',
+                title : 'Episode '+ chapter+' [SUB]',
+                language : 'SUB',
+            });
+        });
+        let rawchapters = data.fetch[0].show.availableEpisodesDetail.raw;
+        rawchapters.forEach(chapter => {
+            chapterlist.push({
+                id : '/watch/'+mangaid+'/episode-'+chapter+'-raw',
+                title : 'Episode '+ chapter+' [RAW]',
+                language : 'RAW',
+            });
+        });
+        let dubchapters = data.fetch[0].show.availableEpisodesDetail.dub;
+        dubchapters.forEach(chapter => {
+            chapterlist.push({
+                id : '/watch/'+mangaid+'/episode-'+chapter+'-dub',
+                title : 'Episode '+ chapter+' [DUB]',
+                language : 'DUB',
+            });
+        });
+        return chapterlist;
+    }
+    async _getPages(chapter) {
+        let request = new Request(new URL(chapter.id, this.url), this.requestOptions);
+        let script = `
+        new Promise(resolve => {
+            let pages = __NUXT__;
+            resolve(pages);
+        });
+        `;
+        let data = await Engine.Request.fetchUI(request, script);
+        let sourcesArray = data.fetch[0].episodeSelections;
+        let goodSource = null;
+        sourcesArray.every(source => {
+            if (source.sourceName.indexOf('Default') > -1) {
+                goodSource = source;

```

Comment:
Why are you using every  if you're not using the result ? Are you just trying to find the sourcesArray.findLast() ?
it could be simplified to goodSource = sourcesArray.findLast(source => source.sourceName.indexOf('Default') > -1);

Reply:
I was drunk i guess. Find is enough.